plugins/uberdev/lib/report_primitives.py

```
@@ -0,0 +1,73 @@
+Deliberately NOT here (pipeline-local): SEV_RANK maps, norm()/fingerprint()/
+dedupe(), load_findings()/read_global(), per-pipeline main()/column schemas.
+"""
+import re
+
+# The findings-to-issues envelope close marker. A finding whose text contains
+# this literal would otherwise close the envelope early and promote
+# attacker-derived rows to trusted prose (security.md #6, MEDIUM, D7). We
+# neutralize it by splitting the closing `</...>` so the literal tag never
+# appears verbatim in output, while staying human-readable.
+_ENVELOPE_CLOSE = "</external-untrusted-input>"
+_ENVELOPE_CLOSE_NEUTRALIZED = "<​/external-untrusted-input>"  # ZWSP after '<'
```

Ruff (0.15.13)

[error] Invalid unescaped character zero-width-space, use "\u200B" instead

Replace with escape sequence

(PLE2515)